### PR TITLE
Adding readyapi4j-maven-plugin

### PR DESCRIPTION
**Pull request by prakashmrec**
I will extend it for local execution. So far I have moved the plugin from a separate repo to readyapi4j and updated it to use latest version of readyapi4j libraries.

I have added a maven-plugin-tester module as well to test the maven plugin as part of the build.

### CODE REVIEW

File: modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+    @Parameter
+    private String callback;
+
+    @Parameter
+    private boolean disableFiltering;
+
+    @Parameter(defaultValue = "true")
+    private boolean failOnFailures;
+
+    @Parameter(required = true, property = "ready-api-test-server.username")
```
Comment (olensmar) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: suggest we use the same name as the system-properties use in the facade;
- testserver.endpoint
- testserver.user
- testserver.password

File: modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+
+    @Parameter(defaultValue = "${project.basedir}/src/test/resources/recipes", required = true)
+    private File recipeDirectory;
+
+    @Parameter(defaultValue = "${project.basedir}/target/test-recipes", required = true)
+    private File targetDirectory;
+
+    @Parameter(defaultValue = "${basedir}/target/surefire-reports")
+    private File reportTarget;
+
```
Comment (olensmar) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: missing a property for server repository!?

Comment (prakashmrec) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: Repository support will require some more changes. I would recommend to do that in a separate PR.

**Comment by olensmar** [comment]
@prakashmrec wasn't there a readme that we could move here also?

**Comment by prakashmrec** [comment]
Yes, there was. Let me add that.

**Comment by olensmar** [comment]
thanks!
